src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            addIfFirstNode(value);
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            newNode.next = head;
```

Comment:
```suggestion
        }
        if (index == 0) {
            newNode.next = head;
```
so we will get rid of the if else construction